lib/hex_web/owners.ex

```diff
@@ -0,0 +1,62 @@
+defmodule HexWeb.Owners do
+  import Ecto
+  alias HexWeb.{User, Package, Repo}
+
+  def all(package) do
+    assoc(package, :owners)
+    |> Repo.all
+  end
+
+  def find(email) do
+    Repo.get_by!(User, email: email)
+  end
+
+  def add(package, owner, opts \\ []) do
+    multi =
+      Ecto.Multi.new
+      |> Ecto.Multi.insert(:owner, Package.build_owner(package, owner))
+      |> audit_fun(opts)
+
+    case HexWeb.Repo.transaction(multi) do
+      {:ok, _} ->
+        owners = all(package)
+
+        HexWeb.Mailer.send(
+          "owner_add.html",
+          "Hex.pm - Owner added",
+          Enum.map(owners, & &1.email),
+          username: owner.username,
+          email: owner.email,
+          package: package.name
+        )
+        :ok
+
+      {:error, :owner, changeset, _} ->
+        {:error, changeset}
+    end
+  end
+
+  def remove(package, owner, owners \\ nil, opts \\ []) do
+    owners = owners || all(package)
+
+    {:ok, _} =
+      Ecto.Multi.new
+      |> Ecto.Multi.delete_all(:package_owner, Package.owner(package, owner))
+      |> audit_fun(opts)
+      |> HexWeb.Repo.transaction
+
+    HexWeb.Mailer.send(
+      "owner_remove.html",
+      "Hex.pm - Owner removed",
+      Enum.map(owners, & &1.email),
+      username: owner.username,
+      email: owner.email,
+      package: package.name
+    )
+  end
+
+  defp audit_fun(multi, opts) do
+    noop = & &1
+    Keyword.get(opts, :audit_fun, noop).(multi)
```

Comment:
perhaps would be better to do `Keyword.fetch`, and if someone doesn't want stuff audited would explicitly pass noop fun, not sure.